lxd/container_lxc.go

```diff
@@ -1377,6 +1377,11 @@ func (c *containerLXC) initLXC() error {
 				if err != nil {
 					return err
 				}
+			} else if m["nictype"] == "vfio" {
+				err = lxcSetConfigItem(cc, fmt.Sprintf("%s.%d.link", networkKeyPrefix, networkidx), m["host_name"])
+				if err != nil {
+					return err
+				}
```

Comment:
Why are we still doing this?

Reply:
In case we're already running. Initlxc should be called in such codepaths

Reply:
I guess that's fine. It's confusing because when looking at the code, you see us set it twice. It may also end up with an invalid config, albeit temporarily.

If I was to "lxc init" a container, then add a vfio device. There would be no pre-existing host_name value in volatile, so initLXC would then put an empty string in config. Whether not having the line at all would be better, I don't know.

Reply:
I'm not sure either. I'd say we leave it in for now and if this causes trouble we will remove it. We will need to rework a bunch of that logic anyway when `liblxc 3.0` lands.